### PR TITLE
docs: move server recipes to how-to guides

### LINKED ISSUE #6107: docs: move Server recipes to how-to guides
At the moment, the [Server](https://loopback.io/doc/en/lb4/Server.html) page in "Behind the scenes" is mixing [explanations](https://documentation.divio.com/explanation/) with [how-to guides](https://documentation.divio.com/how-to-guides/). This makes it difficult for our users to find the relevant content (a recipe to achieve the outcome they want).

As part of our adoption of the four-quadrant documentation system (see #5549 and #5718), we should extract all guides from explanation pages into guide pages (see #5768 for inspiration) and keep explanation pages focused on explaining the concepts & providing background information.

This story is a part of Epic #5783 move recipes to how-to guides. 

## Acceptance criteria

Move the following recipes into "How-to guides", decide yourself whether to create a new page for each recipe or group multiple recipes in one page. For each recipe moved elsewhere, add a new entry to "Common task" list in the "Server" page.

- [ ] [Configure REST API Explorer](https://loopback.io/doc/en/lb4/Server.html#configure-the-api-explorer), including [Disable redirect to API Explorer](https://loopback.io/doc/en/lb4/Server.html#disable-redirect-to-api-explorer) and [Use a self-hosted API Explorer](https://loopback.io/doc/en/lb4/Server.html#use-a-self-hosted-api-explorer)
- [ ] [Customize CORS](https://loopback.io/doc/en/lb4/Server.html#customize-cors)
- [ ] [Customize express settings](https://loopback.io/doc/en/lb4/Server.html#express-settings)
- [ ] [Configure the base path](https://loopback.io/doc/en/lb4/Server.html#configure-the-base-path)
- [ ] [Configure the router](https://loopback.io/doc/en/lb4/Server.html#configure-the-router)
- [ ] [Configure the request body parser options](https://loopback.io/doc/en/lb4/Server.html#configure-the-request-body-parser-options)
- [ ] [Add servers to application instance](https://loopback.io/doc/en/lb4/Server.html#add-servers-to-application-instance)
- [ ] [Enhance OpenAPI Specification](https://loopback.io/doc/en/lb4/Server.html#enhance-openapi-specification)

### PR DESCRIPTION
**Pull request by mrmodise**
Closes #6107

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Customizing-Server-configuration.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
+---
```
Comment (dhmlau) [inline docs/site/Customizing-Server-configuration.md:1]: Suggest to rename the file name (just changing case) to be `Customizing-server-configuration.md` to be consistent with other file naming convention. In this case, we might need to change the permalink to be consistent as well. Thanks.

**Review by dhmlau — COMMENTED** [review]
@mrmodise, thanks for improving the docs! I have some minor comments. Thanks.

File: docs/site/sidebars/lb4_sidebar.yml [hunk 2]
```diff
@@ -198,6 +198,10 @@ children:
     output: 'web, pdf'
     children:
 
+    - title: 'Customizing Server configuration'
```
[the remark this answers is not in the document]
Comment (mrmodise) [reply docs/site/sidebars/lb4_sidebar.yml:201]: Agreed

File: docs/site/Customizing-Server-configuration.md (hunk 1 above)
Comment (mrmodise) [reply docs/site/Customizing-Server-configuration.md:1]: Done

**Review by dhmlau — APPROVED** [review]

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**